### PR TITLE
Fix branch not changing

### PR DESCRIPTION
**Pull request by shadmansaleh**
#37 
git branch wasn't changing because git command was being executed in current directory rather then file directory .

I donn't know what the async code actually doing . So there might be a better way to solve it.

### CODE REVIEW

**Comment by hoob3rt** [comment]
What you did is also done by `set autochdir`. But I think this is not an optimal solution, although it fixes the issue of branch not changing, we don't want to set global vim variables for users.

That's not the only issue tho, even with your patch if you open a file in git directory and switch to a file outside of a git directory the branch persists.

There might be a fix with `on_stderr` from `async`, you can try to have fun with it.

**Comment by shadmansaleh** [comment]
>That's not the only issue tho, even with your patch if you open a file in git directory and switch to a file outside of a git directory the branch persists.

Oh I didn't notice that before . I've added a stderr detection now branch gets removed when not in a git-repo.

File: lua/lualine/components/branch.lua [hunk 1]
```diff
@@ -8,12 +8,22 @@ local get_git_branch = async:new({
...
+        git_branch = ''
+      end
+    end
+  end,
 })
 
 local timer = vim.loop.new_timer()
 timer:start(0, 1000, vim.schedule_wrap(function()
+  local cur_dir = vim.fn.getcwd()
+  vim.api.nvim_set_current_dir(vim.fn.expand("%:p:h"))
```
Comment (hoob3rt) [inline lua/lualine/components/branch.lua:24]: This should be error checked, right now it produces errors in `fzf` in floating mode, probably due to temp buffers or sth.
pic-window-210108-1015-58

Comment (hoob3rt) [reply lua/lualine/components/branch.lua]: Okay I resolved this, should be working now.

**Comment by shadmansaleh** [comment]
Is there any other exception you can think of ?

**Comment by hoob3rt** [comment]
> Is there any other exception you can think of ?

I changed string matching to pcall, will handle all exceptions now. I think this is done, let me know what you think

**Comment by shadmansaleh** [comment]
I think pcall is the way to go.

**Pull request merged**